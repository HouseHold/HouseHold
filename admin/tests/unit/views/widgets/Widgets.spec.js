import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Widgets from '@/views/widgets/Widgets'

Vue.use(CoreuiVue)

describe('Widgets.vue', () => {
  it('has a name', () => {
    expect(Widgets.name).toBe('Widgets')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Widgets)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Widgets', () => {
    const wrapper = shallowMount(Widgets)
    expect(wrapper.is(Widgets)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Widgets)
    expect(wrapper.element).toMatchSnapshot()
  })
})
