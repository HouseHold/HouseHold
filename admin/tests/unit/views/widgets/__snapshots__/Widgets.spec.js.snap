// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets.vue renders correctly 1`] = `
<div>
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        footer="Lorem ipsum dolor sit amet enim."
        value="25"
      >
        <div
          class="h4 m-0"
        >
          89.9%
        </div>
         
        <div
          class="card-header-actions"
        >
          <a
            class="card-header-action position-absolute"
            href="https://coreui.io/vue/docs/components/widgets"
            rel="noreferrer noopener"
            style="right: 5px; top: 5px;"
            target="_blank"
          >
            <small
              class="text-muted"
            >
              docs
            </small>
          </a>
        </div>
         
        <div>
          Lorem ipsum...
        </div>
         
        <cprogress-stub
          class="progress-xs my-3 mb-0"
          color="success"
          max="100"
          precision="0"
          value="25"
        />
      </cwidgetprogress-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        color="info"
        footer="Lorem ipsum dolor sit amet enim."
        header="12.124"
        text="Lorem ipsum..."
        value="25"
      />
    </ccol-stub>
     
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        color="warning"
        footer="Lorem ipsum dolor sit amet enim."
        header="$98.111,00"
        text="Lorem ipsum..."
        value="25"
      />
    </ccol-stub>
     
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        color="danger"
        footer="Lorem ipsum dolor sit amet enim."
        header="2 TB"
        text="Lorem ipsum..."
        value="25"
      />
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        color="success"
        footer="Lorem ipsum dolor sit amet enim."
        header="89.9%"
        inverse="true"
        text="Lorem ipsum..."
        value="25"
      />
    </ccol-stub>
     
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        color="info"
        footer="Lorem ipsum dolor sit amet enim."
        header="12.124"
        inverse="true"
        text="Lorem ipsum..."
        value="25"
      />
    </ccol-stub>
     
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        color="warning"
        footer="Lorem ipsum dolor sit amet enim."
        header="$98.111,00"
        inverse="true"
        text="Lorem ipsum..."
        value="25"
      />
    </ccol-stub>
     
    <ccol-stub
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgetprogress-stub
        color="danger"
        footer="Lorem ipsum dolor sit amet enim."
        header="2 TB"
        inverse="true"
        text="Lorem ipsum..."
        value="25"
      />
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="primary"
        header="$1.999,50"
        iconpadding="true"
        text="Income"
      >
        <cicon-stub
          name="cil-settings"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="info"
        header="$1.999,50"
        iconpadding="true"
        text="Income"
      >
        <cicon-stub
          name="cil-laptop"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="warning"
        header="$1.999,50"
        iconpadding="true"
        text="Income"
      >
        <cicon-stub
          name="cil-moon"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="danger"
        header="$1.999,50"
        iconpadding="true"
        text="Income"
      >
        <cicon-stub
          name="cil-bell"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="primary"
        header="$1.999,50"
        text="Income"
      >
        <cicon-stub
          name="cil-settings"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="info"
        header="$1.999,50"
        text="Income"
      >
        <cicon-stub
          name="cil-laptop"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="warning"
        header="$1.999,50"
        text="Income"
      >
        <cicon-stub
          name="cil-moon"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="3"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="danger"
        header="$1.999,50"
        text="Income"
      >
        <cicon-stub
          name="cil-bell"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      col="12"
      lg="4"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="primary"
        header="$1.999,50"
        text="Income"
      >
        <cicon-stub
          class="mx-5 "
          name="cil-settings"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="4"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="info"
        header="$1.999,50"
        text="Income"
      >
        <cicon-stub
          class="mx-5 "
          name="cil-laptop"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
     
    <ccol-stub
      col="12"
      lg="4"
      sm="6"
      tag="div"
    >
      <cwidgeticon-stub
        color="warning"
        header="$1.999,50"
        text="Income"
      >
        <cicon-stub
          class="mx-5 "
          name="cil-moon"
          width="24"
        />
      </cwidgeticon-stub>
    </ccol-stub>
  </crow-stub>
   
  <widgetsbrand-stub
    nocharts="true"
  />
   
  <widgetsbrand-stub />
   
  <ccardgroup-stub
    class="mb-4"
    tag="div"
  >
    <cwidgetprogressicon-stub
      color="info"
      header="87.500"
      text="Visitors"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-people"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      color="success"
      header="385"
      text="New Clients"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-userFollow"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      color="warning"
      header="1238"
      text="Products sold"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-basket"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      header="28%"
      text="Returning Visitors"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-chartPie"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      color="danger"
      header="5:34:11"
      text="Avg. Time"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-speedometer"
      />
    </cwidgetprogressicon-stub>
  </ccardgroup-stub>
   
  <ccardgroup-stub
    class="mb-4"
    tag="div"
  >
    <cwidgetprogressicon-stub
      color="info"
      header="87.500"
      inverse="true"
      text="Visitors"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-people"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      color="success"
      header="385"
      inverse="true"
      text="New Clients"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-userFollow"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      color="warning"
      header="1238"
      inverse="true"
      text="Products sold"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-basket"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      color="primary"
      header="28%"
      inverse="true"
      text="Returning Visitors"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-chartPie"
      />
    </cwidgetprogressicon-stub>
     
    <cwidgetprogressicon-stub
      color="danger"
      header="5:34:11"
      inverse="true"
      text="Avg. Time"
      value="25"
    >
      <cicon-stub
        height="36"
        name="cil-speedometer"
      />
    </cwidgetprogressicon-stub>
  </ccardgroup-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="info"
        header="87.500"
        text="Visitors"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-people"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="success"
        header="385"
        text="New Clients"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-userFollow"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="warning"
        header="1238"
        text="Products sold"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-basket"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="primary"
        header="28%"
        text="Returning Visitors"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-chartPie"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="danger"
        header="5:34:11"
        text="Avg. Time"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-speedometer"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="info"
        header="972"
        text="comments"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-speech"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
  </crow-stub>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="info"
        header="87.500"
        inverse="true"
        text="Visitors"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-people"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="success"
        header="385"
        inverse="true"
        text="New Clients"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-userFollow"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="warning"
        header="1238"
        inverse="true"
        text="Products sold"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-basket"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="primary"
        header="28%"
        inverse="true"
        text="Returning Visitors"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-chartPie"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="danger"
        header="5:34:11"
        inverse="true"
        text="Avg. Time"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-speedometer"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
     
    <ccol-stub
      md="2"
      sm="6"
      tag="div"
    >
      <cwidgetprogressicon-stub
        color="info"
        header="972"
        inverse="true"
        text="comments"
        value="25"
      >
        <cicon-stub
          height="36"
          name="cil-speech"
        />
      </cwidgetprogressicon-stub>
    </ccol-stub>
  </crow-stub>
   
  <widgetsdropdown-stub />
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      lg="2"
      sm="4"
      tag="div"
    >
      <cwidgetsimple-stub
        header="title"
        text="1,123"
      >
        <cchartlinesimple-stub
          backgroundcolor="transparent"
          bordercolor="danger"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          style="height: 40px;"
        />
      </cwidgetsimple-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="2"
      sm="4"
      tag="div"
    >
      <cwidgetsimple-stub
        header="title"
        text="1,123"
      >
        <cchartlinesimple-stub
          backgroundcolor="transparent"
          bordercolor="primary"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          style="height: 40px;"
        />
      </cwidgetsimple-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="2"
      sm="4"
      tag="div"
    >
      <cwidgetsimple-stub
        header="title"
        text="1,123"
      >
        <cchartlinesimple-stub
          backgroundcolor="transparent"
          bordercolor="success"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          style="height: 40px;"
        />
      </cwidgetsimple-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="2"
      sm="4"
      tag="div"
    >
      <cwidgetsimple-stub
        header="title"
        text="1,123"
      >
        <cchartbarsimple-stub
          backgroundcolor="danger"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          style="height: 40px;"
        />
      </cwidgetsimple-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="2"
      sm="4"
      tag="div"
    >
      <cwidgetsimple-stub
        header="title"
        text="1,123"
      >
        <cchartbarsimple-stub
          backgroundcolor="primary"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          style="height: 40px;"
        />
      </cwidgetsimple-stub>
    </ccol-stub>
     
    <ccol-stub
      lg="2"
      sm="4"
      tag="div"
    >
      <cwidgetsimple-stub
        header="title"
        text="1,123"
      >
        <cchartbarsimple-stub
          backgroundcolor="success"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          style="height: 40px;"
        />
      </cwidgetsimple-stub>
    </ccol-stub>
  </crow-stub>
</div>
`;
